packages/yew/src/platform/rt_wasm_bindgen/time.rs

```diff
@@ -0,0 +1,75 @@
+use std::cell::Cell;
+use std::future::Future;
+use std::pin::Pin;
+use std::rc::Rc;
+use std::task::{Context, Poll};
+use std::time::Duration;
+
+use futures::stream;
+use futures::stream::Stream;
+use gloo::timers::callback::Timeout;
+
+#[inline(always)]
+pub(crate) fn sleep(dur: Duration) -> impl Future<Output = ()> {
+    pub struct Sleep {
+        inner: Option<Timeout>,
+        dur_left: Option<u128>,
+        timeout_registered: Rc<Cell<bool>>,
+    }
+
+    impl Future for Sleep {
+        type Output = ();
+
+        fn poll(mut self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Self::Output> {
+            static I32_MAX_U128: u128 = 2_147_483_647;
+            static I32_MAX_U32: u32 = 2_147_483_647;
+
+            // If polling before the registered timeout is reached, return Pending.
+            if self.timeout_registered.get() {
+                return Poll::Pending;
+            }
+
+            // set_timeout can only accept maximum of i32, so we wrap around if it gets longer.
+            let next_timeout = match self.dur_left.map(|m| (m, u32::try_from(m))) {
+                Some((m_u128, Err(_))) => {
+                    self.dur_left = Some(m_u128 - I32_MAX_U128);
+                    I32_MAX_U32
+                }
+                Some((m_u128, _)) if m_u128 > I32_MAX_U128 => {
+                    self.dur_left = Some(m_u128 - I32_MAX_U128);
+                    I32_MAX_U32
+                }
+                Some((_, Ok(m_u32))) => {
+                    self.dur_left = None;
+                    m_u32
+                }
+                None => return Poll::Ready(()),
+            };
+
+            let waker = cx.waker().clone();
+            self.timeout_registered.set(true);
+            let timeout_registered = self.timeout_registered.clone();
+
+            self.inner = Some(Timeout::new(next_timeout, move || {
+                timeout_registered.set(false);
+                waker.wake();
+            }));
+
+            Poll::Pending
+        }
+    }
+
+    Sleep {
+        inner: None,
+        dur_left: Some(dur.as_millis()),
+        timeout_registered: Cell::new(false).into(),
+    }
```

Comment:
Why implement it here? What's wrong with the `gloo` implementation? If there's a bug or improvement to be made, I think we should fix it there.

Reply:
See: https://github.com/rustwasm/gloo/issues/121

The difficulty of patching it with gloo is due to its callback-first API.

Patching `gloo_timers` requires recursively calling `setTimeout`, which is difficult to implement due to ownership rules.
It gets worse with `Interval`, as we have to implement it with a series of patched `Timeout` than using `setInterval`.
At that time, it's simply easier to implement Interval with `Stream` and make the API future-first.
However, this is a bigger change than it is needed to patch around this behaviour.

I patched it here because we have a future-first API and this have very little impact on code size.

Reply:
That makes sense. I'm not sure who actually is using a timer that's over 24 days long but I'm fine with merging this change.

Patching it in Gloo would be needless (and not small) breaking change